codegen/generate.pl

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/perl
+#
+# Vulpes (c) 2019 gbMichelle
+#
+# This program is free software under the GNU General Public License v3.0 or later. See LICENSE for more information.
+#
+
+use strict;
+use warnings;
+use File::Basename qw( dirname basename );
+use YAML::XS qw( LoadFile );
+
+use lib dirname(__FILE__); # Include own directory
+use CodeGen::Signature qw( yaml_signatures_to_cpp_definitions );
+
+my $input_file_count = scalar @ARGV;
+my $i = 0;
+
+foreach my $filename (@ARGV) {
+    # Terminal output non-sense.
+    $i++;
+    my $progress = int(($i*100)/$input_file_count);
+    printf "[%3d%s] Building CPP files for %s\n", $progress, "%", $filename;
+
+    # Load yaml definition.
+    my $file = LoadFile($filename);
+
+    # Get path without file extension.
+    my $output_stem = $filename;
+    $output_stem =~ s/\.\w+$//;
+
+    # Open new source and headers for writing.
+    open(OUTPUT_SRC,  ">$output_stem.cpp");
+    open(OUTPUT_HEAD, ">$output_stem.hpp");
+
+    # basename with yaml extension
+    my $yaml_basename = basename($filename);
+
+    my $license_header = qq{/*
```

Comment:
You should define this as a subroutine (bootleg constant) at the top of this script instead.